app/controllers/questions_controller.rb

```diff
@@ -70,11 +137,25 @@ def log_time
   end
 
   def next
-    setup = Setup.take
-    week = setup.weeks.find_by_number(params[:week_number])
-    RecommenderSystem::Recommender.setup(current_user.id,week.id)
-    best = RecommenderSystem::HybridRecommender.new.get_best
-    los = LearningObject.find(best[0])
-    redirect_to action: "show", id: los.url_name
+    room = Room.find_by_id(params[:room_number])
+
+    if current_user.show_solutions
+      # V rezime so zobrazovanim spravnych odpovedi su otazky zoradene podla id
+      lo_id = params[:lo_id]
+      los = room.learning_objects.where("learning_object_id > ?",lo_id).order(id: :asc).first
+      if los.nil?
+        los = room.learning_objects.order(id: :asc).first
+      end
+    else
+      # V rezime bez zobrazovania spravnych odpovedi sa vyberie otazka, z tych, ktore este nevidel, resp. ak taka nie je tak nahodna z danej miestnosti
+      los_not_visited = room.get_dont_visited
+      if los_not_visited.nil?
```

Comment:
Naozaj tu mas moznost dostat `nil`? Lebo podla mna to bude skor prazdny vysledok a nie objekt `nil`.

Ak tam nedostanes `nil`, mozes to kludne prerobit takto:
```
los = room.get_dont_visited.first || room.learning_objects.shuffle.first
```